mesonbuild/wrap/wrap.py

```diff
@@ -566,18 +566,21 @@ def get_file_internal(self, what: str) -> str:
 
             return path.as_posix()
 
+    def unpack_archive(self, path):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            shutil.unpack_archive(path, tmpdir)
+            src = os.listdir(tmpdir)
+            if len(src) != 1:
+                raise WrapException("Error patch archive format.")
```

Comment:
Last small nitpicking: The error message could be a bit more informative. Should tell we expect the archive to contain only 1 top directory.

Reply:
@xclaesse  error message == "Archive must contain only one top directory" ok ?